apps/google-analytics-4/lambda/src/app.ts

```diff
@@ -8,21 +9,46 @@ import {
   GoogleApiServerError,
 } from './services/google-api';
 import { ServiceAccountKeyFile } from './types';
+import { verifySignedRequestMiddleware } from './lib/verify-signed-request-middleware';
+import { InvalidSignature } from './lib/errors/invalid-signature';
+import { UnableToVerifyRequest } from './lib/errors/unable-to-verify-request';
+
+dotenv.config(); // TODO: load env vars from .env.local
+
 const app = express();
 
+// allow all OPTIONS requests
+app.options('/*', function (req, res, _next) {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE,OPTIONS');
+  res.header(
+    'Access-Control-Allow-Headers',
+    'Content-Type, Authorization, Content-Length, X-Requested-With'
+  );
+  res.send(200);
+});
+// validate signed requests
+app.use(['/api/credentials', '/api/account_summaries'], verifySignedRequestMiddleware);
+
 // Maps an error class name -> a handler function that takes the error of that type as input and returns
 // a correctly configured API error as output.
 const errorClassToApiErrorMap: ApiErrorMap = {
   GoogleApiError: (e: GoogleApiError) => new ApiError(e.details, e.name, e.httpStatus),
   GoogleApiClientError: (e: GoogleApiClientError) => new ApiError(e.details, e.name, e.httpStatus),
   GoogleApiServerError: (e: GoogleApiServerError) => new ApiError(e.details, e.name, e.httpStatus),
+  InvalidSignature: (e: InvalidSignature) => new ApiError(e.message, e.name, 403),
+  UnableToVerifyRequest: (e: UnableToVerifyRequest) => new ApiError(e.message, e.name, 422),
```

Comment:
Per my comment on the frontendcode, both of these errorTypes should be handled explicitly inside our `makeSignedRequest` function (I believe)

Reply:
i believe this is obsolete now – discussed on slack w/u :)